articles/azure-functions/durable/durable-functions-entities.md

```diff
@@ -139,21 +201,185 @@ public static async Task Run(
+
+The functions `SignalEntityAsync<TInterfaceType>`, `SignalEntity<TInterfaceType>`, and `CreateProxy<TInterfaceType>` all require a first argument of type `EntityId` to specify the target entity, which includes both the entity name and the entity key.
+
+However, since it can get tedious to construct these entity ids, we support a shorter form specifically for situations where there is single, unique entity function that implements `TInterfaceType`, and accept a first argument of type string (just the entity key) instead of EntityId (the entire entity id). If a unique implementation cannot be found at runtime, `InvalidOperationException` will be thrown. 
+
+### Rules for Entity Interfaces
+
+As with all interfaces in .NET, both of these are allowed:
+* an entity can implement multiple interfaces
+* an interface can be implemented by multiple entities
+
+This allows entities to serve multiple roles, and allows entity communication patterns to be implemented as reusable libraries.
+
+However, since each method in an entity interface must represent a valid entity operation, we do impose some rules:
+* Entity interfaces must only define methods.
+* Entity interface methods must not define more than one parameter.
+* Entity interface method parameters must be JSON-serializable objects or values.
+* Entity interface methods must return `void`, `Task`, or `Task<T>` where `T` is a JSON-serializable object or value.
+
+If any of these rules are violated, an `InvalidOperationException` will be thrown at runtime when the interface is used as a type argument to `SignalEntity` or `CreateProxy`. The exception message will explain which rule was broken.
+
+> [!NOTE]
+Interface methods returning `void` can only be signaled (one-way), not called (two-way). Interface methods returning `Task` or `Task<T>` can be either called or signalled. If called, they return the result of the operation, or re-throw exceptions thrown by the operation. However, when signalled, they do not return the actual result or exception from the operation, but just the default value.
+
+## On the Class-Based Syntax
+
+The class-based model is similar to the programming model popularized by [Orleans](https://www.microsoft.com/research/project/orleans-virtual-actors/). In this model, an entity type is defined as a .NET class. Each method of the class is an operation that can be invoked by an external client. Unlike Orleans, however, .NET interfaces are optional. The previous *Counter* example did not use an interface, but it can still be invoked via other functions or via HTTP API calls.
+
+### Custom initialization on first access
+
+To construct an entity object in a special manner the very first time it is accessed, add a conditional before the `DispatchAsync`:
+
+```csharp
+[FunctionName(nameof(Counter))]
+public static Task Run([EntityTrigger] IDurableEntityContext ctx)
+{
+    if (!ctx.HasState)
+    {
+        ctx.SetState(...);
+    }
+    ctx.DispatchAsync<Counter>();
+}
+```
```

Comment:
Everything from line 253 to here also looks like it belongs in the **durable-functions-bindings.md** article since it's not really conceptual. That's where we currently talk SDK-specific things. If we think there's too much content there already then we may want to create a new language-specific article for the entity programming models.